### PR TITLE
HPCC-27299 WsLogAccess reports logline count

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds loglinescount and totallineavailable resp fields

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
Sample req/resp:

```
{
 "GetLogsRequest": {
   "LogCategory": "1",
   "SearchByValue": "W20220427-200649",
   "Range": {
     "StartDate": "2022-03-02T01:26:27",
     "EndDate": ""
   },
   "LogLineLimit": 1,
   "LogLineStartFrom": 0,
   "SelectColumnMode": "1",
   "Columns": {
     "Item": [
       
     ]
   },
   "Format": "JSON"
 }
}
```

```
{
 "GetLogsResponse": {
   "LogLines": "{\"lines\": [{\"fields\": [ {\n \"hpcc.log.threadid\": \"9\",\n \"hpcc.log.timestamp\": \"2022-04-27 20:06:52.673\",\n \"kubernetes.container.name\": \"hthor-w20220427-200649\",\n \"hpcc.log.jobid\": \"W20220427-200649\",\n \"hpcc.log.audience\": \"OPR\",\n \"hpcc.log.message\": \"hthor build community_8.6.23-closedown0Debug[tags/community_8.6.22-rc1-0-gf4178b-dirty]\",\n \"hpcc.log.sequence\": \"00000001\",\n \"hpcc.log.class\": \"INF\",\n \"hpcc.log.procid\": \"9\"\n} ]}]}",
   "LogLineCount": 1,
   "TotalLogLinesAvailable": 87
 }
}
```

File: system/jlib/jlog.cpp [hunk 1]
```diff
@@ -3458,36 +3458,37 @@ ILogAccessFilter * getBinaryLogAccessFilterOwn(ILogAccessFilter * arg1, ILogAcce
...
+logQueryResultDetails fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, const StringArray & cols, LogAccessLogFormat format)
 {
     LogAccessConditions logFetchOptions;
     logFetchOptions.setTimeRange(timeRange);
     logFetchOptions.setFilter(filter);
     logFetchOptions.copyLogFieldNames(cols); //ensure these fields are declared in m_logMapping->queryProp("WorkUnits/@contentcolumn")? or in LogMap/Fields?"
-
-    return logAccess.fetchLog(logFetchOptions, returnbuf, format);
+//Rodrigo, update this function signature
+logQueryResultDetails totals = logAccess.fetchLog(logFetchOptions, returnbuf, format);
```
Comment (wangkx) [inline system/jlib/jlog.cpp:3468]: minor: wrong indent

File: esp/services/ws_logaccess/WsLogAccessService.cpp [hunk 2]
```diff
@@ -143,7 +143,13 @@ bool Cws_logaccessEx::onGetLogs(IEspContext &context, IEspGetLogsRequest &req, I
     logFetchOptions.setStartFrom((offset_t)startFrom);
 
     StringBuffer logcontent;
-    m_remoteLogAccessor->fetchLog(logFetchOptions, logcontent, logAccessFormatFromName(req.getFormat()));
+    logQueryResultDetails logQueryResultDetails = m_remoteLogAccessor->fetchLog(logFetchOptions, logcontent, logAccessFormatFromName(req.getFormat()));
```
Comment (wangkx) [inline esp/services/ws_logaccess/WsLogAccessService.cpp:146]: logQueryResultDetails &logQueryResultDetails?

File: system/jlib/jlog.cpp [hunk 3]
```diff
@@ -3458,36 +3458,37 @@ ILogAccessFilter * getBinaryLogAccessFilterOwn(ILogAccessFilter * arg1, ILogAcce
 // LOG ACCESS HELPER METHODS
 
 // Fetches log entries - based on provided filter, via provided IRemoteLogAccess instance
-bool fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, const StringArray & cols, LogAccessLogFormat format)
+logQueryResultDetails fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, const StringArray & cols, LogAccessLogFormat format)
 {
```
Comment (wangkx) [inline system/jlib/jlog.cpp:3462]: logQueryResultDetails &?
the same at other function declarations below.

File: system/jlib/jlog.hpp [hunk 4]
```diff
@@ -1592,14 +1592,20 @@ interface IRemoteLogAccessStream : extends IInterface
     virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead) = 0;
 };
 
+struct logQueryResultDetails
```
Comment (wangkx) [inline system/jlib/jlog.hpp:1595]: LogQueryResultDetails?

**Review by wangkx — COMMENTED** [review]
@rpastrana a few comments/questions

**Comment by rpastrana** [comment]
@wangkx decided to restructure a bit. Please review latest changes.

**Review by wangkx — APPROVED** [review]
@rpastrana looks fine

**Comment by rpastrana** [comment]
@GordonSmith @jeclrsg any comments on this change before we finalize it?

**Comment by jeclrsg** [comment]
@rpastrana Looks good to me

**Review by ghalliday — APPROVED** [review]

**Pull request merged**